.github/workflows/e2e_and lighthouse_performance_tests.yml

```diff
@@ -134,15 +134,54 @@ jobs:
             ${{ runner.os }}-build-${{ env.cache-name }}-
             ${{ runner.os }}-build-
             ${{ runner.os }}-
-      - name: Download build files artifact
+      - name: Install Oppia Dependencies
+        uses: ./.github/actions/install-oppia-dependencies
+      - name: Install chrome
+        run: python -m scripts.install_chrome_for_ci
+      - name: Install ffmpeg
+        run: sudo apt install ffmpeg
+      - name: Attempt to download build files artifact
+        id: download_artifact_1
         uses: actions/download-artifact@v3
+        continue-on-error: true
         with:
           name: build_files
           path: /home/runner/work/oppia/
+      - name: (Artifact re-build) Cache third_party/static
+        if: steps.download_artifact_1.outcome == 'failure'
+        uses: actions/cache@v3
+        env:
+          cache-name: cache-node-modules
+        with:
+          path: /home/runner/work/oppia/oppia/third_party/static
+          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('dependencies.json') }}
+          restore-keys: |
+            ${{ runner.os }}-build-${{ env.cache-name }}-
+            ${{ runner.os }}-build-
+            ${{ runner.os }}-
+      - name: (Artifact re-build) Build Webpack
+        if: steps.download_artifact_1.outcome == 'failure'
+        run: python -m scripts.build --prod_env
+      - name: (Artifact re-build) Zip build files
+        if: steps.download_artifact_1.outcome == 'failure'
+        run: |
+          zip -rqy build_files.zip oppia/third_party oppia_tools oppia/build oppia/webpack_bundles oppia/proto_files oppia/app.yaml oppia/assets/hashes.json oppia/proto_files oppia/extensions/classifiers/proto/* oppia/backend_prod_files oppia/dist
+        working-directory: /home/runner/work/oppia
+      - name: (Artifact re-build) Upload build files 
+        if: steps.download_artifact_1.outcome == 'failure'
+        uses: actions/upload-artifact@v3
+        with:
+          name: build_files
+          path: /home/runner/work/oppia/build_files.zip
+          retention-days: 1
+      - name: (artifact re-build) Remove build_files.zip
+        if: steps.download_artifact_1.outcome == 'failure'
+        run: rm build_files.zip
       - name: Unzip build files
+        if: steps.download_artifact_1.outcome != 'failure'
         run: |
           ls -la
-          unzip build_files.zip
+          unzip -o build_files.zip
```

Comment:
Hmm. I think this reveals that I made a mistake in my earlier review. Sorry about that! It looks like we need to run the `Install Oppia Dependencies` step at different times depending on whether we can use the cache. If we can use the cache, then we run it after downloading the cache. If we cannot use the cache, then we have to run it earlier so that we can build the cache. ffmpeg can still be installed before we deal with caching, and I think we can put the install Chrome step after all the caching logic (it has to be later since we need the Python installed during `Install Oppia Dependencies`). Does that make sense? If I made another mistake, please let me know!

With the current code, we'll do some unnecessary dependency installation, which could be handled by the cache, even when the cache is available.

Reply:
I see, so we want to install dependencies after unzipping so that we don't re-install when we do the force-unzip. I'll update that in the code!